indirect.h

```diff
@@ -343,4 +347,30 @@ struct std::hash<xyz::indirect<T, Alloc>> {
   }
 };
 
+#if (__cpp_lib_format >= 201907L)
+
+namespace xyz {
+
+template <class T>
+concept is_formattable = requires(T t) { std::formatter<T, char>{}; };
```

Comment:
Since C++23 we have the concept [std::formattable<T, charT>](http://eel.is/c++draft/format.formattable#concept:formattable) which does exactly this. But I am not sure which standard libraries already implement this.

Reply:
Looks like this is unsupported in Visual Studio yet, but is supported in GCC 13 onward and Clang 17 onwards.